### PR TITLE
Implement initial PoC for ApiServerError

### PR DESCRIPTION
**Pull request by hatchan**
Initial overhaul of errors in fpx:
- Returning ApiServerError from a handler 
- Add DbError abstraction for database error (mimics the one from fp/api)
- Add ApiClientError for the API client

Add simple ApiClient

Change casing, tagging for all API models. 

Resolves: FP-3786

### CODE REVIEW

File: fpx/src/api/client.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    client: reqwest::Client,
+    base_url: Url,
+}
+
+impl ApiClient {
+    pub fn new(client: reqwest::Client, base_url: Url) -> Self {
+        Self { client, base_url }
+    }
+
+    pub async fn do_req<T, E>(&self, method: Method, path: String) -> Result<T, ApiClientError<E>>
```
Comment (mellowagain) [inline fpx/src/api/client.rs]: can we have `path` as `impl Into<String>` to allow passing `&str`?

**Review by mellowagain — APPROVED** [review]

File: fpx/src/api/client.rs (hunk 1 above)
Comment (hatchan) [reply fpx/src/api/client.rs]: Good point. I actually made it a `impl AsRef<str>` as `join` only requires a `&str`.

**Review by stephlow — APPROVED** [review]

File: fpx/src/api/client.rs [hunk 2]
```diff
@@ -11,11 +14,18 @@ pub struct ApiClient {
 }
 
 impl ApiClient {
-    pub fn new(client: reqwest::Client, base_url: Url) -> Self {
+    /// Create a new ApiClient with a default reqwest::Client.
+    pub fn new(base_url: Url) -> Self {
+        let client = reqwest::Client::new();
```
Comment (mellowagain) [inline fpx/src/api/client.rs]: do we wanna set a user agent htat identifies us?

**Review by mellowagain — APPROVED** [review]

File: fpx/src/api/client.rs (hunk 2 above)
Comment (hatchan) [reply fpx/src/api/client.rs]: Good idea

**Pull request merged**